shuffle/cli/src/shared.rs

```diff
@@ -771,49 +793,58 @@ mod test {
         fs::create_dir_all(dir.path().join(".shuffle")).unwrap();
         home.write_default_networks_config_into_toml().unwrap();
         let networks_cfg = home.read_networks_toml().unwrap();
-        assert_eq!(networks_cfg, NetworksConfig::default());
+        assert_eq!(networks_cfg, NetworksConfig::default().unwrap());
     }
 
-    fn get_test_network() -> Network {
+    fn get_test_localhost_network() -> Network {
         Network {
             name: "localhost".to_string(),
-            base: "http://127.0.0.1".to_string(),
-            json_rpc_port: 8080,
-            dev_api_port: 8081,
+            json_rpc_url: Url::from_str("http://127.0.0.1:8080").unwrap(),
+            dev_api_url: Url::from_str("http://127.0.0.1:8080").unwrap(),
+            faucet_url: None,
+        }
+    }
+
+    fn get_test_trove_testnet_network() -> Network {
+        Network {
+            name: "trove_testnet".to_string(),
+            json_rpc_url: Url::from_str("http://trove.aws.hlw3truzy4ls.com").unwrap(),
+            dev_api_url: Url::from_str("https://api.trove.aws.hlw3truzy4ls.com").unwrap(),
+            faucet_url: Some(Url::from_str("https://trove.aws.hlw3truzy4ls.com").unwrap()),
         }
     }
 
     #[test]
     fn test_generate_default_networks_config() {
         let mut network_map = BTreeMap::new();
-        network_map.insert("localhost".to_string(), get_test_network());
+        network_map.insert("localhost".to_string(), get_test_localhost_network());
+        network_map.insert(
+            "trove_testnet".to_string(),
+            get_test_trove_testnet_network(),
+        );
         let networks_cfg = NetworksConfig {
             networks: network_map,
         };
-        assert_eq!(NetworksConfig::default(), networks_cfg);
+        assert_eq!(NetworksConfig::default().unwrap(), networks_cfg);
```

Comment:
Shouldn't need an unwrap() here imo. cc @bmwill

Reply:
Yeah if you have a method named `default` you should be implementing the `Default` trait who's signature does not return an error.